scripts/skills/perks/perk_rf_phalanx.nut

```diff
@@ -45,45 +51,40 @@ this.perk_rf_phalanx <- ::inherit("scripts/skills/skill", {
 
 	function onAnySkillUsed( _skill, _targetEntity, _properties )
 	{
-		if (_targetEntity != null && _skill.isAttack() && !_skill.isRanged())
+		if (_skill.isAttack() && !_skill.isRanged() && (this.m.RequiredDamageType == null || _skill.getDamageType().contains(this.m.RequiredDamageType)))
 		{
-			_properties.Reach += this.getCount();
+			_properties.Reach += this.getAllyCount();
```

Comment:
This does not currently use your new `ReachModifier` variable.
I recommend adding a new `getReachModifier()` function, which utilizes `getAllyCount` internally.

Then you can use that also for the `isHidden` and `getTooltip` implementations

Reply:
I've changed it as you have suggested. It is the most "robust" implementation but I'd like to ask is it really worth it? Now we have 2x the performance hit in `isHidden()`. I'd say it is safe to assume that this perk works always with ally count by virtue of its design, so the `ReachModifier` and `StartSurroundCountAtModifier` being member variables and `isHidden()` only checking for `getAllyCount()` seems clean enough and more efficient to me.